### ISSUE TITLE
[docs] [proposal] Expand rule metadata to include code examples

### ISSUE THREAD

**Issue by aervin** [palantir/tslint#3601]
Users have [requested](https://github.com/palantir/tslint/issues/3594) that the rule docs be updated with examples of correct/incorrect code (code that will pass linting/code that will fail linting).

I propose that rule metadata be expanded to include optional code example props:
```ts
export interface IRuleMetadata {
    /* ... existing props*/

    codeExamplePass?: string;
    codeExampleFail?: string;
    codeExampleFix?: string;
}
```
or perhaps
```ts
export interface IRuleMetadata {
    /* ... existing props */

    codeExamples?: {
        pass: string;
        fail: string;
        fix?: string;
    }
}
```
From there, it's small updates to `buildDocs.ts` and `rule.html`, is that right?

**Comment by Shinigami92** [comment]
Can you give these two lines a div container with css-class?
Then you can style them with in example:
```css
.fail { background-color: #ffdcdc; }
.pass { background-color: #dcffdc; }
```
[aervin/tslint => /docs/_layouts/rule.html L50](https://github.com/aervin/tslint/blob/57c418861fc7c7baff2fbea0f9d0edd2fc9fc74f/docs/_layouts/rule.html#L50)
[aervin/tslint => /docs/_layouts/rule.html L53](https://github.com/aervin/tslint/blob/57c418861fc7c7baff2fbea0f9d0edd2fc9fc74f/docs/_layouts/rule.html#L53)

**Comment by aervin** [comment]
@Shinigami92 Is something like this what you had in mind? 
![tslint-code-examples](https://user-images.githubusercontent.com/13973880/34917182-ec8c5502-f910-11e7-9550-f99f6ca564b1.png)

**Comment by Shinigami92** [comment]
Looks great :)
